master/master.hpp

```diff
@@ -26,29 +26,36 @@
 #include "base/hash.hpp"
 #include "core/hash_ring.hpp"
 #include "core/zmq_helpers.hpp"
-#include "master/assigners.hpp"
 
 namespace husky {
 
 class Master {
    public:
-    Master();
-    virtual ~Master();
+    static Master& get_instance() {
+        static Master master;
+        return master;
+    }
 
+    void setup();
+    virtual ~Master();
     void init_socket();
     void serve();
-
     void handle_message(uint32_t message, const std::string& id);
-
     inline int get_num_machines() const { return num_machines; }
     inline zmq::socket_t* get_socket() const { return resp_socket; }
     inline const std::string& get_cur_client() const { return cur_client; }
 
-    inline void register_handler(uint32_t msg_type, std::function<void()> handler) {
-        external_handlers[msg_type] = handler;
+    inline void register_main_handler(uint32_t msg_type, std::function<void()> handler) {
+        external_main_handlers[msg_type] = handler;
+    }
+
+    inline void register_setup_handler(std::function<void()> handler) {
```

Comment:
Could you run `scripts/clang-format.sh` and then commit again? There're some differences output in [travis ci](https://travis-ci.org/husky-team/husky/builds/161270459)